### PR TITLE
Fix stream reader raw data read and add skip samples method

### PR DESCRIPTION
**Pull request by mnovakDS**
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:
 - [x] Pull request title reflects its content
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

# Description:

### CODE REVIEW

File: core/opendaq/reader/include/opendaq/stream_reader.h [hunk 1]
```diff
@@ -78,6 +78,16 @@ DECLARE_OPENDAQ_INTERFACE(IStreamReader, ISampleReader)
...
+
+    // [arrayArg(count, 1)]
+    /*!
+     * @brief Skips the specified amount of samples.
+     *
+     * @param[in,out] count The maximum amount of samples to be skipped. If the `count` is less than
+     * available the parameter value is set to the actual amount and only the available
+     * samples are skipped. The rest of the buffer is not modified or cleared.
+     */
+    virtual ErrCode INTERFACE_FUNC skipSamples(SizeT* count) = 0;
```
Comment (JakaMohorkoDS) [inline core/opendaq/reader/include/opendaq/stream_reader.h:90]: If we have `read` without "Samples", this should also simply be `skip`.

File: core/opendaq/reader/src/stream_reader_impl.cpp [hunk 2]
```diff
@@ -548,6 +565,27 @@ ErrCode StreamReaderImpl::readWithDomain(void* samples,
...
+    info.prepare(nullptr, *count, milliseconds(0));
+    if (info.dataPacket.assigned())
+        errCode = readPacketData();
+
+    if (OPENDAQ_SUCCEEDED(errCode) && info.remainingToRead <= *count)
+        errCode = readPackets(nullptr);
+
+    *count = *count - info.remainingToRead;
+    return errCode;
+}
```
Comment (JakaMohorkoDS) [inline core/opendaq/reader/src/stream_reader_impl.cpp:587]: This must return `IReaderStatus`, or you will end up dropping events, making them irrecoverable.

File: core/opendaq/reader/include/opendaq/stream_reader.h (hunk 1 above)
Comment (JakaMohorkoDS) [inline core/opendaq/reader/include/opendaq/stream_reader.h:90]: Python bindings also need to be updated on an interface change.

**Review by JakaMohorkoDS — CHANGES_REQUESTED** [review]

File: core/opendaq/reader/include/opendaq/stream_reader.h (hunk 1 above)
Comment (mnovakDS) [reply core/opendaq/reader/include/opendaq/stream_reader.h:90]: I followed the interface naming of multi reader. Do I also change it there?

**Review by JakaMohorkoDS — APPROVED** [review]

**Review by denise-opendaq — APPROVED** [review]

**Pull request merged**